src/Discord.hs

```diff
@@ -116,18 +117,24 @@ runDiscord opts = do
 -- | Runs the main loop 
 runDiscordLoop :: DiscordHandle -> RunDiscordOpts -> IO T.Text
 runDiscordLoop handle opts = do
-  resp <- liftIO $ writeRestCall (discordHandleRestChan handle) GetCurrentUser
+  resp <- startupRestCalls
   case resp of
     Left (RestCallInternalErrorCode c e1 e2) -> libError $
              "HTTP Error Code " <> T.pack (show c) <> " " <> TE.decodeUtf8 e1
                                                    <> " " <> TE.decodeUtf8 e2
     Left (RestCallInternalHttpException e) -> libError ("HTTP Exception -  " <> T.pack (show e))
     Left (RestCallInternalNoParse _ _) -> libError "Couldn't parse GetCurrentUser"
```

Comment:
this error is technically incomplete now, as the Left value could also be from current application. no big issue though

Reply:
Generalized the error. 

Added issue for specific ADT https://github.com/discord-haskell/discord-haskell/issues/192